WordPress/openverse
# Avoid emulation for `ov` development environment

## Fixes

Fixes #4934 by @dhruvkb 
Addresses #4944 by @d3jawu 

## Problem

I was trying to use the `ov` environment more proactively in anticipation of the upcoming WordCamp. During this I noticed that `ov` was considerably slower than it should've been when running under emulation on macOS. This PR aims to fix that.

## Description



This PR allows us to go back to running `ov` in the architecture native to the host machine instead of emulating for `linux/amd64`. This change significantly improves performance.

For example, it would take many minutes for linting to set up and run earlier, which now only takes a few seconds. This makes commits and pushes so much more frictionless.

To do this,
- we replaced the `vale` Python package with a simpler download script that automatically downloads architecture specific binaries from GH releases. With this change, we can also drop the entire Python project we had scaffolded inside `.vale/` for one dependency.
- we [made `k6` optional](https://github.com/WordPress/openverse/issues/4934#issuecomment-2352492106) in the Dockerfile. It will automatically be installed when the architecture is supported and will be skipped otherwise. If we find ourselves needing it more, we can switch to a method similar to Vale above and get it from GH releases.

## Testing Instructions


0. Check out the PR.
1. If you were using `ov` before, clean it up:
   - `ov clean`
   - `ov init`
2. Check that workflows continue to work normally, for example linting:
   ```bash
   ov just lint
   ```

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
